packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4008,7 +4008,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   }
 
   void _stopCursorBlink({ bool resetCharTicks = true }) {
-    _cursorBlinkOpacityController.value = 0.0;
+    _cursorBlinkOpacityController.value = renderEditable.floatingCursorOn ? 1.0 : 0.0;
```

Comment:
Why is this needed? Shouldn't `_stopCursorBlink` stop the cursor blink as expected without changes?

Reply:
Even when the floating cursor is displayed, we cannot ignore the call to `_stopCursorBlink` in `_didChangeTextEditingValue`, which leads to some test failures. I haven't investigated the specific reasons yet, but I think it's feasible to always keep the cursor visible in `_stopCursorBlink` when the floating cursor appears. Should I investigate the reasons for the test failures or maintain the status quo?

Reply:
I think I've found some clues. We need to set the value of `_cursorBlinkOpacityController` to stop the animation. This is to avoid re-executing `_onCursorTick` when the animation ends at this location: https://github.com/flutter/flutter/blob/5887d6c74248fc6542a49e4a999b09b2aa58ef9c/packages/flutter/lib/src/widgets/editable_text.dart#L4004.

Reply:
Sorry for the delay on the review, just ran this locally and I understand what you mean. I'm okay with the code as-is then, but could you add a comment explaining why we take the floating cursor into account.

Reply:
Done.